test/unittests/execute_floating_point_test.cpp

```diff
@@ -102,23 +102,27 @@ TYPED_TEST(execute_floating_point_types, nan_matchers)
 
     EXPECT_THAT(Void, Not(CanonicalNaN(TypeParam{})));
     EXPECT_THAT(Trap, Not(CanonicalNaN(TypeParam{})));
-    EXPECT_THAT(ExecutionResult{Value{TypeParam{}}}, Not(CanonicalNaN(TypeParam{})));
-    EXPECT_THAT(ExecutionResult{Value{FP::nan(FP::canon)}}, CanonicalNaN(TypeParam{}));
-    EXPECT_THAT(ExecutionResult{Value{-FP::nan(FP::canon)}}, CanonicalNaN(TypeParam{}));
-    EXPECT_THAT(ExecutionResult{Value{FP::nan(FP::canon + 1)}}, Not(CanonicalNaN(TypeParam{})));
-    EXPECT_THAT(ExecutionResult{Value{-FP::nan(FP::canon + 1)}}, Not(CanonicalNaN(TypeParam{})));
-    EXPECT_THAT(ExecutionResult{Value{FP::nan(1)}}, Not(CanonicalNaN(TypeParam{})));
-    EXPECT_THAT(ExecutionResult{Value{-FP::nan(1)}}, Not(CanonicalNaN(TypeParam{})));
+    EXPECT_THAT(ExecutionResult{TypedValue{TypeParam{}}.value}, Not(CanonicalNaN(TypeParam{})));
```

Comment:
What's the point here of constructing `TypedValue` and immediately discarding the type part?

Reply:
There exists `TypedValue(FP)` but not `Value(FP)` so the `TypedValue(FP).value` is a workaround for the missing `Value(FP)`.

The `TypedValue(FP)` was added to create `TypedValue` type without going though floating-point object.